core/domain/summary_services.py

```diff
@@ -68,9 +68,25 @@ def get_displayable_exp_summary_dicts_matching_ids(exploration_ids):
     Please use this function when needing summary information to display on
     exploration summary tiles in the frontend.
     """
-    displayable_exp_summaries = []
     exploration_summaries = (
         exp_services.get_exploration_summaries_matching_ids(exploration_ids))
+
+    return get_displayable_exp_summary_dicts(exploration_summaries)
+
+
+def get_displayable_exp_summary_dicts(exploration_summaries):
+    """Given a list of exploration summary domain objects, returns
+     a list of the human-readable exploration summary dicts that
+     correspond to explorations that are currently non-private
+     and not deleted.
+    """
+    displayable_exp_summaries = []
```

Comment:
I suggest moving this line to just before the "for ind, exploration_summary" loop, since it's not used at all in the next few lines. Then you have better structure: the first part of this function computes the view_counts, and the rest computes the displayed_exp_summaries.